pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -209,15 +227,25 @@ func (k *k8sClient) GetClusterStatus() (clusterStatus *ClusterStatus, err error)
 	return clusterStatus, nil
 }
 
-func (k *k8sClient) UpdateReplicas(expReplicas int32) (prevRelicas int32, err error) {
-	prevRelicas, err = k.updateReplicasAppsV1(expReplicas)
+func (k *k8sClient) UpdateReplicas(expReplicas int32) (err error) {
+	for _, target := range k.target.targets {
+		_, err := k.UpdateTargetReplicas(expReplicas, target, k.target.namespace)
+		if err == nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (k *k8sClient) UpdateTargetReplicas(expReplicas int32, target target, namespace string) (prevRelicas int32, err error) {
```

Comment:
No need to pass the namespace, you already have it stored in `k. scaleTargets.namespace`. Same applies for other methods too.